code/modules/projectiles/gun.dm

```diff
@@ -312,11 +313,28 @@
 
 		if(muzzle_flash)
 			set_light(muzzle_flash)
-
+	//We've successfully fired a shot! Now we add recoil, and add a delay to remove the recoil (2s, affected by vig)
+	recoil += recoil_buildup*10 //Course wanted noticeable recoil. EX: 0.2 buildup * 10 = 2
+	var/skill_offset = user.stats.getStat(STAT_VIG)/50 //For example, IH have 40 VIG so this translates into a 0.8s cooldown reduction,meaning they can fire double as fast. Feel free to change this, Course!
+	var/recoil_reset_time = 20 //After you fire your shot, you must wait 2 seconds for it to become accurate again
+	if(skill_offset > 0) //Fuck negative numbers
```

Comment:
Why forbid negatives from affecting those? If you got your VIG in the red, it makes sense for you to suck at recoil control.